docs/Argo-CD-Integration.md

```diff
@@ -1,26 +1,25 @@
 
 Thus the ApplicationSet controller:
+
 - Does not create/modify/delete Kubernetes resources (other than the `Application` CR)
 - Does not connect to clusters other than the one Argo CD is deployed to
 - Does not interact with namespaces other than the one Argo CD is deployed within
 
-The ApplicationSet controller can thus be thought of as an `Application` 'factory', taking an `ApplicationSet` as input, and generating as output one or more Argo CD `Application` resources that correspond to the parameters of that set.
+It is Argo CD itself that is responsible for the actual deployment of the generated child `Application` resources, such as Deployments, Services, and ConfigMaps.
 
-![ApplicationSet controller vs Argo CD, interaction diagram](assets/Argo-CD-Integration/ApplicationSet-Argo-Relationship-v2.png)
+The ApplicationSet controller can thus be thought of as an `Application` 'factory', taking an `ApplicationSet` resource as input, and outputting one or more Argo CD `Application` resources that correspond to the parameters of that set.
 
-In this diagram, an `ApplicationSet` resource exists, and it is the ApplicationSet controller that is responsible for creating the corresponding `Application` resources. 
+![ApplicationSet controller vs Argo CD, interaction diagram](assets/Argo-CD-Integration/ApplicationSet-Argo-Relationship-v2.png)
 
-The resulting `Application` resources are then managed Argo CD: that is, Argo is responsible for actually deploying the child resources. 
+In this diagram an `ApplicationSet` resource is defined, and it is the responsibility of the ApplicationSet controller to create the corresponding `Application` resources. The resulting `Application` resources are then managed Argo CD: that is, Argo CD is responsible for actually deploying the child resources. 
 
-Argo CD generates the application's Kubernetes resources based on to the contents of the Git repository defined within the Application `spec`, deploying e.g. Deployments, Service, and other resources.
+Argo CD generates the application's Kubernetes resources based on the contents of the Git repository defined within the Application `spec` field, deploying e.g. Deployments, Service, and other resources.
 
-Creation, updates, or deletions of ApplicationSets will have a direct effect on the Applications present in the Argo CD namespace. Likewise, cluster events (the addition/deletion of Argo CD cluster secrets, when using Cluster generator), or changes in Git (when using Git generator), will be used as input to the ApplicationSet controller in constructing `Application` resources.
+Creation, updates, or deletion of ApplicationSets will have a direct effect on the Applications present in the Argo CD namespace. Likewise, cluster events (the addition/deletion of Argo CD cluster secrets, when using Cluster generator), or changes in Git (when using Git generator), will be used as input to the ApplicationSet controller in constructing `Application` resources.
```

Comment:
Should "updates" be "update" (singular)?

Reply:
Done.